MyToDo/App/Filters/ToDoFilter.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  ToDoFilter.swift
+//  MyToDo
+//
+//  Created by Ivan Ruiz Monjo on 23/2/21.
+//
+
+import SwiftUI
+
+final class ToDoFilter: ObservableObject {
```

Comment:
I don't think all of these properties need to be `@Published`. In fact, this implementation is open to some bugs. I can do this now:

```
filter.upcoming = false
filter.categories = []
filter.hasFilters = true
```

`hasFilters` can be a computed property since, ideally, it's only set in response to one of your `@Published` properties changing. So this should work fine:

```
var hasFilters: Bool { upcoming || !categories.isEmpty }
```

Reply:
I see, for some reason I thought hasFilters it won't be re-read. (Yours was my first approach). I see know what you are saying, if upcoming or categories changes, View will be re-rerendered and hasFilters will be also re-read